src/entities/Task/model/slice/tasksSlice.ts

```diff
@@ -0,0 +1,87 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import {
+    StatusEnum, SubTask, Task, TasksSchema,
+} from '../types/task';
+
+const initialState: TasksSchema = {
+    data: [],
+};
+
+const updateLocalStorage = (tasks: Task[]) => {
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+const getTaskById = (tasks: (Task | SubTask)[], id: string) => tasks.find((task) => task.id === id);
+
+export const tasksSlice = createSlice({
+    name: 'tasks',
+    initialState,
+    reducers: {
+        getAllTasks: (state) => {
+            state.data = JSON.parse(localStorage.getItem('tasks') || '[]');
+        },
+
+        createNewTask: (state, action: PayloadAction<Task>) => {
+            state.data.push(action.payload);
+            updateLocalStorage(state.data);
+        },
+
+        changeTaskStatus: (state, action: PayloadAction<Task>) => {
+            const { status } = action.payload;
+
+            const currentTask = getTaskById(state.data, action.payload.id) as Task;
+            currentTask.status = status;
+
+            if (status === StatusEnum.COMPLETED) {
+                currentTask.completedAt = Date.now();
+                currentTask.subTasks = currentTask.subTasks.map((sub) => (
```

Comment:
you can update subTasks with `.forEach()` instead of creating new subTasks with `.map()`

Reply:
Done